diff --git a/src/crash_handler/crash_handler.zig b/src/crash_handler/crash_handler.zig
index a193d21..2670729 100644
--- a/src/crash_handler/crash_handler.zig
+++ b/src/crash_handler/crash_handler.zig
@@ -882,7 +882,7 @@ fn handleSegfaultPosix(sig: i32, info: *const std.posix.siginfo_t, ctx_ptr: ?*co
             const fp = uctx.mcontext.regs[29];
             faulting_pc = pc;
 
-            const writer = std.io.getStdErr().writer();
+            const writer = std.debug.getStdErr().writer();
             writer.writeAll("\n============ CRASH REGISTER DUMP (aarch64) ============\n") catch {};
             writer.print("Faulting PC:  0x{x:0>16}\n", .{pc}) catch {};
             writer.print("Link Register (LR/x30): 0x{x:0>16}\n", .{lr}) catch {};
diff --git a/src/resolver/resolver.zig b/src/resolver/resolver.zig
index c3d96c0..84c572f 100644
--- a/src/resolver/resolver.zig
+++ b/src/resolver/resolver.zig
@@ -2860,7 +2860,7 @@ pub const Resolver = struct {
         var _safe_path: ?string = null;
 
         // Start at the top.
-        'queue_loop: while (queue_slice.len > 0) {
+        queue_loop: while (queue_slice.len > 0) {
             var queue_top = queue_slice[queue_slice.len - 1];
             defer parent_result = queue_top.result;
             queue_slice.len -= 1;
@@ -2914,7 +2914,7 @@ pub const Resolver = struct {
                     error.EACCES, error.PermissionDenied => {
                         if (!is_target_dir) {
                             parent_result = missing_parent_result;
-                            continue :'queue_loop;
+                            continue :queue_loop;
                         }
                         r.dir_cache.markNotFound(queue_top.result);
                         const cached_dir_entry_result = rfs.entries.getOrPut(queue_top.unsafe_path) catch unreachable;
diff --git a/src/workaround_missing_symbols.zig b/src/workaround_missing_symbols.zig
index a5a7605..754096a 100644
--- a/src/workaround_missing_symbols.zig
+++ b/src/workaround_missing_symbols.zig
@@ -9,15 +9,15 @@ pub const linux = if (bun.Environment.isAndroid) struct {
     // section with sh_addralign=64.
 
     pub const lstat = blk: {
-        const T = *const fn ([*:0]const u8, *std.os.linux.Stat) callconv(.C) c_int;
+        const T = *const fn ([*:0]const u8, *std.os.linux.Stat) callconv(.c) c_int;
         break :blk @extern(T, .{ .name = "lstat" });
     };
     pub const fstat = blk: {
-        const T = *const fn (c_int, *std.os.linux.Stat) callconv(.C) c_int;
+        const T = *const fn (c_int, *std.os.linux.Stat) callconv(.c) c_int;
         break :blk @extern(T, .{ .name = "fstat" });
     };
     pub const stat = blk: {
-        const T = *const fn ([*:0]const u8, *std.os.linux.Stat) callconv(.C) c_int;
+        const T = *const fn ([*:0]const u8, *std.os.linux.Stat) callconv(.c) c_int;
         break :blk @extern(T, .{ .name = "stat" });
     };
 } else struct {
