diff --git a/src/bun.js.zig b/src/bun.js.zig
index d79d53e..9eab3aa 100644
--- a/src/bun.js.zig
+++ b/src/bun.js.zig
@@ -396,6 +396,12 @@ pub const Run = struct {
             this.entry_path = vm.transpiler.fs.top_level_dir;
         }
 
+        if (!vm.transpiler.resolver.opts.preserve_symlinks) {
+            if (strings.contains(this.entry_path, "/install/global/node_modules/")) {
+                vm.transpiler.resolver.opts.preserve_symlinks = true;
+            }
+        }
+
         var printed_sourcemap_warning_and_version = false;
 
         if (vm.loadEntryPoint(this.entry_path)) |promise| {
diff --git a/src/cli/run_command.zig b/src/cli/run_command.zig
index 0c171c7..83443af 100644
--- a/src/cli/run_command.zig
+++ b/src/cli/run_command.zig
@@ -1842,6 +1842,21 @@ pub const RunCommand = struct {
                 bun.options.defaultLoaders.get(path.name.ext) orelse .tsx;
             if (loader.canBeRunByBun() or loader == .html or loader == .md) {
                 log("Resolved to: `{s}`", .{path.text});
+                if (!ctx.args.preserve_symlinks and strings.contains(path.text, "/install/global/node_modules/")) {
+                    ctx.args.preserve_symlinks = true;
+                    const preserve_symlinks = this_transpiler.resolver.opts.preserve_symlinks;
+                    defer this_transpiler.resolver.opts.preserve_symlinks = preserve_symlinks;
+                    this_transpiler.resolver.opts.preserve_symlinks = true;
+                    if (this_transpiler.resolver.resolve(
+                        this_transpiler.fs.top_level_dir,
+                        target_name,
+                        .entry_point_run,
+                    ) catch null) |re| {
+                        if (re.path()) |p| {
+                            return _bootAndHandleError(ctx, p.text, loader);
+                        }
+                    }
+                }
                 return _bootAndHandleError(ctx, path.text, loader);
             } else {
                 log("Resolved file `{s}` but ignoring because loader is {s}", .{ path.text, @tagName(loader) });
